src/client/pydaos/raw/daos_api.py

```diff
@@ -1023,7 +1023,10 @@ def insert_array(self, dkey, akey, c_data, txn):
         # self.epoch_range.epr_hi = ~0
 
         self.txn = txn
-        c_tx = ctypes.c_uint64(txn)
+        if txn is not None:
+            c_tx = ctypes.c_uint64(txn)
```

Comment:
this is still wrong.. from what i gather you are setting the transaction value, but in the C api, the tx is just a handle and the assiciated timestamp/uuid is set internally and hidden from the user.

Reply:
Some of the Existing Api call it's being set the new Transaction and later getting commited so I kept it provision instead of removing completely. 

write_an_obj() 
        txn = self.get_new_tx()
        ioreq.single_insert(c_dkey, c_akey, c_value, c_size, txn) (Added provision by default it will be 0 and use the DAOS_TX_NONE) but if user want to get the new tx and pass they can.
        self.commit_tx(txn)